internal/poller/poller.go

```diff
@@ -19,8 +21,15 @@ type Poller struct {
 }
 
 func New(interval time.Duration, pollFunc func() (interface{}, error)) *Poller {
+	wrappedFn := func() (interface{}, error) {
+		defer func() {
+			panics.LogAndPanic(recover(), debug.Stack())
+		}()
+		return pollFunc()
+	}
```

Comment:
Panics can occur in the goroutine where the poll func is started. If we don't want this automatic wrapping we can update each caller of this library to manually inject it into their pollfunc.